src/components/button/button_content.tsx

```diff
@@ -91,11 +91,13 @@ export const OuiButtonContent: FunctionComponent<
       <OuiLoadingSpinner className="ouiButtonContent__spinner" size="m" />
     );
   } else if (iconType) {
+    const { title } = { ...textProps };
     buttonIcon = (
       <OuiIcon
         className="ouiButtonContent__icon"
         type={iconType}
         size={iconSize}
+        title={title}
         color="inherit" // forces the icon to inherit its parent color
       />
     );
```

Comment:
@joshuarrrr,

As requested, I made the changes you suggested. Now, I am passing the title prop inside the definition of the `buttonIcon` variable (an `OuiIcon` component). Now the fix makes more clear the code and avoid creating empty spans in cases where are not button icons